src/js/components/generateDetachedFiles/DetachedFileA.jsx

```diff
@@ -71,19 +78,27 @@ export default class DetachedFileA extends React.Component {
     }
 
     pickedYear(fy) {
+        const fyAvailablePeriods = availablePeriodsInFY(fy);
         this.setState({
+            periodArray: fyAvailablePeriods.periodArray,
+            period: fyAvailablePeriods.period,
             fy
         });
     }
 
-    pickedQuarter(quarter) {
+    pickedPeriod(period) {
         this.setState({
-            quarter
+            period
         });
     }
 
     generate() {
-        this.props.generateFileA(this.state.agency, this.state.codeType, this.state.quarter, this.state.fy);
+        const agencyName = _.find(this.props.agencyList.agencies, { cgac_code: `${this.state.agency}` }).agency_name || 'Unknown Agency';
```

Comment:
We should determine which code to check (`cgac_code` or `frec_code`) based on `this.state.codeType`. An even better solution would be to store the agency name in the state if it's possible to get that when it is selected, to avoid passing around and searching a list of every agency.